reactive/INTRODUCTION_EN.md

```diff
@@ -0,0 +1,153 @@
+	void install(Me me);
+}
+
+// I didn't declare a lot of things, we are doing pseudocode here
+interface Me {
+	// When the cannon finished it's installation, it will notify us and we get an infinite batch of fuses.
+  // Just because it's the future.
+	void onPrepare(FuseBatch fuses);
+	
+	// This is called whenever the cannon launches a cannonball toward your patron.
+	default void onNext(CannonBall ball) {
+		dance();
+	}
+
+	// This is called when the cannon of the future throws an uncaught runtime exception and explodes.
+  // Yes, the futuuuuuuuure still has runtime exceptions :'c
+	default void onError(Throwable t) {
+		runForYourLife();
+	}
+	
+	// This is called when the cannon has no cannonball left to fire.
+	default void onComplete() {
+		if (boss.isAliveAndKicking())
+			timeToGoOnStrike();
+	}
+}
+
+interface FuseBatch {
+	// We kindly ask the cannon to launch `fuseAmount` cannonballs by firing as many fuses.
+	void fire(long fuseAmount);
+	
+	// If you suddendly remember the inherent kindness of your boss, stow your cannon and try to cancel the cannonballs' throwing.
+  // You might miss some fuses though, trampling on them isn't exactly a precise method.
+	void cancel();
+}
+```
+
+Well now that's done, we don't have anything else to do with our cannon.
+So let's haphazardly pick some code from [reactive-streams](http://www.reactive-streams.org/) since it should be related to the official topic of this sheet.
+By the way, reactive-streams is integrated to the jdk since java 9.
+
+> If you don't know of genericity, I advise you to check the [related courses](../java/généricité)
+
+```java
+interface Publisher<T> {
+  // We prepare the subscription, this might take some time
+	void subscribe(Subscriber<? super T> subscriber);
+}
+
+interface Subscriber<T> {
+	// When the Publisher is done preparing the subscription, it calls this method.
+	void onSubscribe(Subscription s);
+
+	// This is called whenever the Publisher emits a new element.
+	void onNext(T t);
+	
+	// This is called if an error occurs while preparing the subscription or generating a new element.
+	void onError(Throwable t);
+	
+	// This is called when the Publisher knows it has no more elements to emit.
+	void onComplete();
+}
+
+interface Subscription {
+	// Kindly asks the Publisher to emit `elementAmount` elements.
+	// The Publisher emits always exactly as many elements as we asked it for, except in certain cases:
+  // • If the cumulated demand overflow the long max number, it can treat it as unbound, effectively turning into a hot source.
+  // (We'll see what this means in later sheets)
+	// • If the Publisher completes (cannot generate anymore element) or errors (sad life), it will call the related method then stop ANY further emission.
+ 	void request(long elementAmount);
+
+  // Asks the Publisher to cancel the subscription, that is to stop emitting elements.
+	// Previous pending requests might still be met.
+  void cancel();
```

Comment:
Weird alternance space/tab
```suggestion
	// • If the cumulated demand overflow the long max number, it can treat it as unbound, effectively turning into a hot source.
	// (We'll see what this means in later sheets)
	// • If the Publisher completes (cannot generate anymore element) or errors (sad life), it will call the related method then stop ANY further emission.
 	void request(long elementAmount);

	// Asks the Publisher to cancel the subscription, that is to stop emitting elements.
	// Previous pending requests might still be met.
  void cancel();
```